src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -30,31 +34,44 @@
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_CREDITS, PREFIX_TAG);
+            ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_TAG);
 
-        Index index;
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty() || argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                EditCommand.MESSAGE_USAGE));
+        }
 
+        ModuleInfoSearcher moduleInfoSearcher = new ModuleInfoSearcher();
+        ModuleCode code = ParserUtil.parseModuleCode(argMultimap.getPreamble());
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            moduleInfoSearcher.searchModule(code.toString());
+        } catch (CommandException e) {
+            throw new ParseException(e.getMessage());
         }
```

Comment:
Actually why do we need to look up the info for the module to be edited ah? I thought we only need to know if that module has been previously added to GradPad, which is already checked by `EditCommand`. Just asking cos these http requests are expensiveee $$$ 😢

Reply:
HAHAHA oh if the requests are $$$ then we can remove them!

I think I did it this way for the purpose of the message to be displayed to the user...
e.g. if the user types in `edit cs2222 c/cs2101`, GradPad will check first if the module exists before even checking if the module is in GradPad's list.

so if:
Module doesnt exist in nusmods -> "Module does not exist"
Module exists, but not in GradPad -> "Module has not been added yet" 

i hope i managed to explain it clearly, but ya i think we can remove if it really affects performance etc.
Should i?